cmd/ocm-backplane/login/login.go

```diff
@@ -208,9 +210,7 @@ func runLogin(cmd *cobra.Command, argv []string) (err error) {
 			// Hibernating, print an error
 			return fmt.Errorf("cluster %s is hibernating, login failed", clusterKey)
 		}
-		// Check API connection with configured proxy
-		err = bpConfig.CheckAPIConnection()
```

Comment:
Any reason to remove `bpConfig.CheckAPIConnection()`, which is ping to api end-point and validates the connection?
https://github.com/openshift/backplane-cli/blob/bcd5b3e43d6d7a8ec3f2654e9cd7e3fd8d4122de/pkg/cli/config/config.go#L92

Reply:
I tired changing the logic to only perform `CheckAPIConnection` only when the error returned is `errLoginConnection` but fails the test cases. 

```
if errors.Is(err, errLoginConnection) {
			connectionErr := bpConfig.CheckAPIConnection()
			if connectionErr != nil {
				return fmt.Errorf("cannot connect to backplane API URL, check if you need to use a proxy/VPN to access backplane: %v", connectionErr)
			}
		}
```

Test result:
```
• Failure [1.378 seconds]
Login command
/Users/feichashao/git/github/backplane-cli/cmd/ocm-backplane/login/login_test.go:28
  check cluster login
  /Users/feichashao/git/github/backplane-cli/cmd/ocm-backplane/login/login_test.go:187
    should fail when BP API timeouts [It]
    /Users/feichashao/git/github/backplane-cli/cmd/ocm-backplane/login/login_test.go:294

    Expected
        <string>: can't login to cluster: unable to connect to backplane api: dial tcp i/o timeout
    to contain substring
        <string>: cannot connect to backplane API URL
```

while I was trying to tweak the test case, I think we can instead skip testing the api connectivity as the error message from `doLogin` has already did a "test" on the apiserver, we can expose the actual network error + a hint about proxy/vpn to make the error logic simpler.

Reply:
I guess this [PR](https://github.com/openshift/backplane-cli/pull/150) fixed the error overwriting issue.